### ISSUE TITLE
How to initialize the dataset to my own data?

### ISSUE THREAD

**Issue by Saharati90** [vlfeat/matconvnet#264]
Hi,

I'm trying to fine-tune the cnn by initialize the network to "imagenet-vgg-f" and train the network on my own dataset by calling cnn_iamgenet.m. Could you please let me know how I can change the following part to load my own dataset if it's possible?

Thanks

"if exist(opts.imdbPath)
  imdb = load(opts.imdbPath) ;
else
  imdb = cnn_imagenet_setup_data('dataDir', opts.dataDir, 'lite', opts.lite) ;
  mkdir(opts.expDir) ;
  save(opts.imdbPath, '-struct', 'imdb') ;
end"

in "cnn_iamgenet.m" and also "cnn_imagenet_setup_data" function.

**Comment by fengyunxiaozi** [comment]
this is easy. you only need to convert your data to "imdb" structure (here I call it imdb structure).  you can refer to mnist example about how to prepare imdb.

**Comment by Saharati90** [comment]
Thanks a lot.
I did the same. but I'm getting this error when try to train the network.
"Error using vl_nnconv
The FILTERS depth does not divide the DATA depth."

Do you think it's impossible to do:
net = load('imagenet-vgg-f.mat');
instead of:
net = cnn_imagenet_init('model', opts.modelType, ...
                        'batchNormalization', opts.batchNormalization, ...
                    'weightInitMethod', opts.weightInitMethod) ;
in network initialization step in "cnn_imagenet.m" ?

**Comment by fengyunxiaozi** [comment]
This error information "The FILTERS depth does not divide the DATA depth." may derive from that the filter dimension (depth) have not matched your image channels (depth). For example, if the filters of the first layer  is M*N*P*Q， then your input data should be X*X*P*X , other layers resemble this situation. You can check it!

**Comment by titubeta** [comment]
I am also getting the same error. 
@fengyunxiaozi: I am using 'imagenet-caffe-ref' to initialize my network. However, my image data is in grayscale. Can this error be related to the fact that my images are grayscale? 

Is there any pretrained model that can deal with gray scale images?
@Saharati90 Did you figure out this problem?

**Comment by virusme** [comment]
@titubeta 
The pre-trained network you are using expects 3-channel input. If your images are grayscale, then replicate the grayscale channel to generate a 3-channel image and feed it to the network.

Hope, this helps.

**Comment by titubeta** [comment]
@virusme : Thanks

**Comment by fengyunxiaozi** [comment]
@titubeta  you can adopt virusme's solution

**Comment by Addhi86** [comment]
@fengyunxiaozi I have made my data in gray scale and stored as 
```
im= imresize ((im2single(rgb2gray(virtual_new_B{i,:}))),[28,28]);
    imdb.images.data(:,:,:, i+Npos) = im;
```
my first filter is 
```
net.layers{end+1} = struct('type', 'conv',...
    'weights',{{f*randn(7,7,1,32, 'single'),zeros(1, 32, 'single')}},...
    'stride', 1,...
    'pad', 0);
```
Still its showing me error 
The FILTERS depth does not divide the DATA depth.

**Comment by zll152** [comment]
hi @Addhi86 ,did U solve it? I have the same problem too.

**Comment by Addhi86** [comment]
@zll152  Yes i have solved that

**Comment by hugolimachaves** [comment]
Hi @Addhi86 
How did you solve that?

**Comment by samo313** [comment]
hello
I am trying to run ImageNet but the following error occurred. how should I resolve this error? any help would be appreciated.

Error using cnn_imagenet_setup_data (line 49)
Make sure that ILSVRC data is correctly installed in

thanks in advance
